src/renderers.rs

```diff
@@ -1,44 +1,37 @@
-use std::fmt;
-use std::rc::Rc;
-use ultraviolet::Mat4;
-use wgpu::{self, Extent3d, TextureView};
-
 use crate::include_spv;
-use crate::render_pass::{BoxedRenderPass, Device, Queue, RenderPass};
+use ultraviolet::Mat4;
+use wgpu::*;
```

Comment:
Please don't import everything from `wgpu`, there are hundreds of exports in that crate. We only use two dozen or so. Name them all explicitly.

Or you can import only the few that are used multiple times, and undo the removal of the full path from most lines in this file for the one-offs.

Reply:
Sure.